lib/workspace.rb

```diff
@@ -0,0 +1,76 @@
+# workspace
+require_relative "user"
+require_relative "channel"
+require "terminal-table"
+require "pry"
+
+class Workspace
+  attr_reader :users, :channels, :selected, :list_users, :list_channels
+
+  def initialize
+    @users = SlackCli::User.list
+    @channels = SlackCli::Channel.list
+    @selected = nil
+  end
+
+  def select_channel(name)
+    @selected = @channels.find do |channel|
+      channel.name == name || channel.slack_id == name
+    end
+    if @selected == nil 
+      puts "That channel does not exist" 
+    else
+      return @selected
+    end
+  end
+
+  def select_user(name)
+    @selected = @users.find do |user|
+      user.name == name || user.slack_id == name
+    end
+
+    if @selected == nil 
+      puts "That user does not exist" 
+    else
+      return @selected
```

Comment:
Since this code is here to interact with the user, I would probably consider it view code and move it to the `main` method.